src/datachain/query/batch.py

```diff
@@ -92,6 +93,86 @@ def __call__(
                 yield [r[0] for r in results] if ids_only else results
 
 
+class DynamicBatch(BatchingStrategy):
```

Comment:
why do we need a whole new strategy? The limit check was already introduced in `query/dataset.py`